src/lookup.rs

```diff
@@ -26,6 +26,7 @@ pub trait LookupParser<'input, Endian>
     /// Parse a header from `input`. Returns a tuple of `input` sliced beyond this header and
     /// all of its entries, `input` sliced to contain just the entries corresponding to this
     /// header (without the header itself), and the parsed representation of the header itself.
+    #[allow(type_complexity)]
```

Comment:
Clippy is right that a number of these are kind of unfortunate, but probably worth dealing with separately.